chaincode/testtuple.go

```diff
@@ -468,6 +468,13 @@ func (testtuple *Testtuple) commitStatusUpdate(db *LedgerDB, testtupleKey string
 		return fmt.Errorf("update testtuple %s failed: %s", testtupleKey, err.Error())
 	}
 
+	// do not update if previous status is already Done or Failed
+	if StatusCanceled == newStatus {
+		if stringInSlice(testtuple.Status, []string{StatusDone, StatusFailed}) {
+			return nil
+		}
```

Comment:
What about also not update a `doing` tuple: as we cannot kill a running task we could let it finish and update their status accordingly.

Reply:
done